narwhal/primary/src/proposer.rs

```diff
@@ -114,6 +119,21 @@ impl Proposer {
         .await;
         debug!("Created {header:?}");
 
+        // Equivocation protection using the proposer store
+        if let Some((_, last_header)) = self.proposer_store.get_last_header()? {
+            if last_header.round == header.round && last_header.epoch == header.epoch {
```

Comment:
what if last_header.epoch != header.epoch? Should we handle all the possibilities and panic for those who are impossible?

Reply:
I thought about doing such checks to ensure the round number is increasing, but its not related what we are trying to accomplish here, so it doesn't seem like the right place for such a check. `last_header.epoch != header.epoch` seems like something that is expected to happen after each epoch change. 

Clearing the proposer store as part of change fragments the logic for correct behavior into two separate locations, opening a possibility for changing one without being aware of the other.

Reply:
thanks @laura-makdah for the reply. Just one note, regarding the proposer store clear, I still believe that doing this as part of the `change_epoch` is safe as everything is happening under the same loop. I mean, if we receive a signal that the epoch has changed, will first do that (clear store), and the continue to next iteration to continue processing (ex create a new header). So anything after that point will only be created for the new epoch - that's why I believe it is a quite safe approach to follow . But really no strong feelings about it.